packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -14,195 +13,95 @@ import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/compone
-        videoWrapperRef.current?.classList.remove(styles['VideoPlayer__container--small'])
-      }
-    }
-
-    const resizeObserver = new ResizeObserver(_ => {
-      if (videoWrapperRef.current) {
-        handleResize()
-      }
-    })
-
-    handleResize()
-
-    const currentRef = videoWrapperRef.current
-
-    resizeObserver.observe(videoWrapperRef.current as Element)
-    return () => {
-      resizeObserver.unobserve(currentRef as Element)
-    }
-  }, [videoWrapperRef, playing])
+  useVideoKeypressHandlers(videoWrapperRef)
 
   return (
     <div
       className={clsx(
         styles.VideoPlayer__container,
         styles.VideoPlayer__overlays,
-        !playing && styles.VideoPlayer__showOverlays,
+        !isPlaying && styles.VideoPlayer__showOverlays,
       )}
       ref={videoWrapperRef}
     >
-      <video
-        ref={videoRef}
-        title={title}
-        controls={false}
-        poster={poster}
-        className={clsx(styles.VideoPlayer, className)}
-        onPlay={handleOnPlay}
-        onPause={handleOnPause}
-        onPlaying={handleOnPlaying}
-        onLoadedMetadata={handleOnLoadedMetadata}
-        onTimeUpdate={handleOnTimeUpdate}
-        {...rest}
-      >
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+      <video ref={ref} title={title} controls={false} className={clsx(styles.VideoPlayer, className)} {...rest}>
         {children}
-        <track kind="captions" />
       </video>
-      {title && (
-        <div className={styles.VideoPlayer__title}>
-          {branding && <MarkGithubIcon size={48} />}
-          <Text size="500" weight="medium" className={styles.VideoPlayer__controlTextColor}>
+      <div className={styles.VideoPlayer__title}>
+        {showBranding && <MarkGithubIcon size={40} />}
+        {showTitle && (
+          <Text size="400" weight="medium" className={styles.VideoPlayer__controlTextColor}>
             {title}
           </Text>
-        </div>
-      )}
-      <button className={styles.VideoPlayer__playButton} onClick={handleVideoPlayback}>
-        {!playing && (
-          <span className={styles.VideoPlayer__playButtonInner}>
-            <svg
-              xmlns="http://www.w3.org/2000/svg"
-              width="24"
-              height="24"
-              viewBox="0 0 24 24"
-              fill="none"
-              aria-hidden="true"
-            >
-              <path
-                d="M1.56489 23.8112C0.884817 24.2389 1.07491e-06 23.7501 1.0398e-06 22.9467L8.28679e-08 1.05473C4.73246e-08 0.241596 0.904067 -0.245404 1.58307 0.201969L18.5829 11.4026C19.2032 11.8113 19.1935 12.7244 18.5647 13.1198L1.56489 23.8112Z"
-                fill="currentColor"
-              />
-            </svg>
-          </span>
         )}
-        <span className="visually-hidden">Play</span>
+      </div>
+      <button
+        className={styles.VideoPlayer__playButton}
+        onClick={() => {
+          isPlaying ? pause() : play()
+        }}
```

Comment:
Seeing this in a few places, but could you please hoist these inline anonymous functions outside of the return please into their own named handlers. They should be passed through a `useCallback` function too.

Reply:
Ah, thank you, I meant to tidy those up.

I've made the changes in 469903a60b588ad03519c97df644a120374c4b9d which ditches the anonymous function entirely so we don't need a useCallback, lmk if that works for you